src/R/Components/Impl/PackageManager/Implementation/RPackageManager.cs

```diff
@@ -68,17 +68,31 @@ internal class RPackageManager : IRPackageManager {
             }
         }
 
-        public async Task UninstallPackageAsync(string name, string libraryPath) {
+        public async Task<PackageLockState> UninstallPackageAsync(string name, string libraryPath) {
             if (!_interactiveWorkflow.RSession.IsHostRunning) {
                 throw new RPackageManagerException(Resources.PackageManager_EvalSessionNotAvailable);
             }
 
             try {
                 using (var request = await _interactiveWorkflow.RSession.BeginInteractionAsync()) {
-                    if (string.IsNullOrEmpty(libraryPath)) {
-                        await request.UninstallPackageAsync(name);
-                    } else {
-                        await request.UninstallPackageAsync(name, libraryPath);
+                    using (var eval = await _interactiveWorkflow.RSession.BeginEvaluationAsync()) {
+                        return await eval.EvaluateAsync<PackageLockState>($"rtvs:::package_uninstall({name.ToRStringLiteral()}, {libraryPath.ToRStringLiteral()})", REvaluationKind.Normal);
+                    }
+                }
+            } catch (RHostDisconnectedException ex) {
```

Comment:
You don't need to catch this exception. I have a changeset that removes such code in other methods